src/qibo/models/parameter.py

```diff
@@ -0,0 +1,42 @@
+"""Model for combining trainable parameters and possible features into circuit parameters."""
+
+
+class Parameter:
+    def __init__(self, func, variational_parameters, features=None):
+        self._variational_parameters = variational_parameters
+        self._featurep = features
+        self.nparams = len(variational_parameters)
+        self.lambdaf = func
+
+    def _apply_func(self, fixed_params=None):
+        params = []
+        if self._featurep is not None:
+            params.append(self._featurep)
+        if fixed_params:
+            params.extend(fixed_params)
+        else:
+            params.extend(self._variational_parameters)
+        return self.lambdaf(*params)
+
+    def _update_params(self, trainablep=None, feature=None):
```

Comment:
I think you don't need this method in case params and features are None, so I would drop the default values.

Reply:
There are cases where we update only the trainable parameters or the feature. If could be for instance that the feature is not embedded in the Parameter lambda function, and I need to prepare for that case as well.